packages/typescript-estree/src/create-program/useProvidedPrograms.ts

```diff
@@ -5,45 +5,49 @@ import * as ts from 'typescript';
 
 import type { ASTAndDefiniteProgram } from './shared';
 import { CORE_COMPILER_OPTIONS, getAstFromProgram } from './shared';
+import { ParseSettings } from '../parseSettings';
 
 const log = debug('typescript-eslint:typescript-estree:useProvidedProgram');
 
-export interface ProvidedProgramsSettings {
-  filePath: string;
-  tsconfigRootDir: string;
-}
-
 function useProvidedPrograms(
   programInstances: Iterable<ts.Program>,
-  { filePath, tsconfigRootDir }: ProvidedProgramsSettings,
+  parseSettings: ParseSettings,
 ): ASTAndDefiniteProgram | undefined {
-  log('Retrieving ast for %s from provided program instance(s)', filePath);
+  log(
+    'Retrieving ast for %s from provided program instance(s)',
+    parseSettings.filePath,
+  );
 
   let astAndProgram: ASTAndDefiniteProgram | undefined;
   for (const programInstance of programInstances) {
-    astAndProgram = getAstFromProgram(programInstance, filePath);
+    astAndProgram = getAstFromProgram(programInstance, parseSettings.filePath);
     // Stop at the first applicable program instance
     if (astAndProgram) {
       break;
     }
   }
 
-  if (!astAndProgram) {
-    const relativeFilePath = path.relative(
-      tsconfigRootDir || process.cwd(),
-      filePath,
-    );
-    const errorLines = [
-      '"parserOptions.programs" has been provided for @typescript-eslint/parser.',
-      `The file was not found in any of the provided program instance(s): ${relativeFilePath}`,
-    ];
-
-    throw new Error(errorLines.join('\n'));
+  if (astAndProgram) {
+    astAndProgram.program.getTypeChecker(); // ensure parent pointers are set in source files
+    return astAndProgram;
   }
 
-  astAndProgram.program.getTypeChecker(); // ensure parent pointers are set in source files
+  const relativeFilePath = path.relative(
+    parseSettings.tsconfigRootDir,
```

Comment:
This used to be `tsconfigRootDir || process.cwd()`, but I'm pretty sure that's unnecessary. `createParseSettings` already defaults `tsconfigRootDir` to `process.cwd()`.